subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/template/KotlinBuildScriptTemplateAdditionalCompilerArgumentsProvider.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:Suppress("DEPRECATION")
```

Comment:
❓ Why is this necessary? If we need it, let's document why (and for how long).

Reply:
Thanks.

🧐 This should not be a Javadoc comment, but instead `//`.

Reply:
Fixed